libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.h

```diff
@@ -121,6 +121,7 @@ class ESP8266WiFiGenericClass {
     public:
         int hostByName(const char* aHostname, IPAddress& aResult);
         int hostByName(const char* aHostname, IPAddress& aResult, uint32_t timeout_ms);
+        int hostByNameAsync(const char* aHostname, IPAddress& aResult, bool* waiting = nullptr);

```

Comment:
Just in general, if `waiting` is `nullptr`, then how would an app actually know that it's done?  I don't think waiting makes sense being optional here.